File: modules/crypto-core/src/main/java/com/shterneregen/securelan/crypto/service/KeyEncodingService.java
package com.shterneregen.securelan.crypto.service;

import javax.crypto.SecretKey;
import java.security.PrivateKey;
import java.security.PublicKey;

public interface KeyEncodingService {
    byte[] encodePublicKey(PublicKey publicKey);

    byte[] encodePrivateKey(PrivateKey privateKey);

    PublicKey decodePublicKey(byte[] encoded);

    PrivateKey decodePrivateKey(byte[] encoded);

    byte[] encodeSecretKey(SecretKey secretKey);

    SecretKey decodeAesKey(byte[] encoded);
}

File: modules/crypto-core/src/main/kotlin/com/shterneregen/securelan/crypto/service/KeyEncodingService.kt
package com.shterneregen.securelan.crypto.service

import java.security.PrivateKey
import java.security.PublicKey
import javax.crypto.SecretKey

interface KeyEncodingService {
    fun encodePublicKey(publicKey: PublicKey): ByteArray

    fun encodePrivateKey(privateKey: PrivateKey): ByteArray
    fun decodePublicKey(encoded: ByteArray): PublicKey
    fun decodePrivateKey(encoded: ByteArray): PrivateKey
    fun encodeSecretKey(secretKey: SecretKey): ByteArray
    fun decodeAesKey(encoded: ByteArray): SecretKey
}
